File: python/01_print_message.py
import threading

def print_message(idx: int):
    thread_id = threading.get_ident()
    print(f"Thread{idx} {thread_id} says hello!")

threads = []
for i in range(10):
    thread = threading.Thread(target=print_message, args=[i])
    threads.append(thread)
    thread.start()

for thread in threads:
    thread.join()

File: C#/01_print_message.cs
using System;
using System.Threading;

class Program {
    static void Main(string[] args) {
        for (int i = 0; i < 10; i++) {
            Thread thread = new Thread(() => {
                Console.WriteLine("Thread {0} is running.", Thread.CurrentThread.ManagedThreadId);
            });
            thread.Start();
        }
    }
}
